src/overlays/actors/ovl_En_Baba/z_en_baba.c

```diff
@@ -86,67 +104,659 @@ static DamageTable D_80BAA468 = {
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Baba/func_80BA9E00.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Baba/func_80BA9E10.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Baba/func_80BA9E48.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Baba/EnBaba_Init.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Baba/EnBaba_Destroy.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Baba/EnBaba_Update.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Baba/func_80BA9FB0.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Baba/func_80BAA198.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Baba/func_80BAA20C.s")
+s32 D_80BAA488[] = {
+    0x0C000119, 0x0A006E14, 0x02000000, 0x1E080F00, 0x1E030400, 0x01050E00, 0x00001E02, 0x05050000,
+};
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Baba/EnBaba_Draw.s")
+s32 D_80BAA4A8[] = { -1, -1, 0 };
+
+u16 D_80BAA4B4[] = { 4000, 4, 1, 3, 6000, 4, 1, 6, 4000, 4, 1, 3, 6000, 4, 1, 6 };
+
+s32 func_80BA8820(EnBaba* this, GlobalContext* globalCtx) {
+    this->unk_144 = (EnOssan*)SubS_FindActor(globalCtx, &this->unk_144->actor, ACTORCAT_NPC, ACTOR_EN_OSSAN);
+
+    if (this->unk_144 != NULL) {
+        return true;
+    }
+    return false;
+}
+
+void func_80BA886C(EnBaba* this, GlobalContext* globalCtx) {
+    Player* player = GET_PLAYER(globalCtx);
+
+    switch (this->unk_1E0) {
+        case 0:
+            if (this->unk_40A & 8) {
+                if (gSaveContext.weekEventReg[33] & 8) {
+                    this->unk_1E0 = 0x2A34;
+                    break;
+                }
+
+                if (gSaveContext.weekEventReg[79] & 0x40) {
+                    this->unk_40A |= 1;
+                    this->unk_1E0 = 0x2A33;
+                    break;
+                }
+                this->unk_40A |= 1;
+                this->unk_1E0 = 0x2A32;
+                break;
+            } else if (player->transformation == PLAYER_FORM_DEKU) {
+                if (!(gSaveContext.weekEventReg[79] & 0x20)) {
+                    gSaveContext.weekEventReg[79] |= 0x20;
+                    this->unk_40A |= 1;
+                    this->unk_1E0 = 0x2A37;
+                    break;
+                } else {
+                    this->unk_40A = this->unk_40A | 1;
+                    this->unk_1E0 = 0x2A38;
+                }
+                break;
+            } else if (!(gSaveContext.weekEventReg[33] & 8)) {
+                if (!(gSaveContext.weekEventReg[73] & 1)) {
+                    this->unk_1E0 = 0x660;
+                    break;
+                }
+                this->unk_1E0 = 0x662;
+                break;
+            } else {
+                if (!(gSaveContext.weekEventReg[73] & 2)) {
+                    this->unk_1E0 = 0x65A;
+                    break;
+                }
+                this->unk_1E0 = 0x65E;
+                break;
+            }
+            break;
```

Comment:
I feel there are way to many `break`'s here. Nearly all of these except the last should be removable with `if-elseif-else` logic

Reply:
Unfortunately not, it changes codegen. If you can find a way to do it let me know, but I can't.